aviatrix/data_source_aviatrix_all_transit_gateways.go

```diff
@@ -516,349 +511,331 @@ func dataSourceAviatrixListAllTransitGateway() *schema.Resource {
-				transitGateway["filtered_spoke_vpc_routes"] = strings.Join(gw.FilteredSpokeVpcRoutes, ",")
+		if len(gw.ExcludeCidrList) != 0 {
+			transitGateway["excluded_advertised_spoke_routes"] = strings.Join(gw.ExcludeCidrList, ",")
+		} else {
+			transitGateway["excluded_advertised_spoke_routes"] = ""
+		}
+
+		gwDetail, err := client.GetGatewayDetail(&gw)
+		if err != nil {
+			return fmt.Errorf("couldn't get Aviatrix Transit Gateway: %s", err)
+		}
+
+		transitGateway["enable_firenet"] = gwDetail.EnableFireNet
+		transitGateway["enable_transit_firenet"] = gwDetail.EnableTransitFireNet
+		transitGateway["enable_egress_transit_firenet"] = gwDetail.EnableEgressTransitFireNet
+		transitGateway["customized_transit_vpc_routes"] = gwDetail.CustomizedTransitVpcRoutes
+
+		if goaviatrix.IsCloudType(gw.CloudType, goaviatrix.AWSRelatedCloudTypes|goaviatrix.AzureArmRelatedCloudTypes|goaviatrix.AliCloudRelatedCloudTypes) && gw.EnableVpcDnsServer == "Enabled" {
+			transitGateway["enable_vpc_dns_server"] = true
+		} else {
+			transitGateway["enable_vpc_dns_server"] = false
+		}
+
+		if gwDetail.EnableAdvertiseTransitCidr == "yes" {
+			transitGateway["enable_advertise_transit_cidr"] = true
+		} else {
+			transitGateway["enable_advertise_transit_cidr"] = false
+		}
+
+		if gwDetail.LearnedCidrsApproval == "yes" {
+			transitGateway["enable_learned_cidrs_approval"] = true
+		} else {
+			transitGateway["enable_learned_cidrs_approval"] = false
+		}
+
+		bgpMSAN := ""
+		for i := range gwDetail.BgpManualSpokeAdvertiseCidrs {
+			if i == 0 {
+				bgpMSAN = bgpMSAN + gwDetail.BgpManualSpokeAdvertiseCidrs[i]
 			} else {
-				transitGateway["filtered_spoke_vpc_routes"] = ""
+				bgpMSAN = bgpMSAN + "," + gwDetail.BgpManualSpokeAdvertiseCidrs[i]
 			}
+		}
+		transitGateway["bgp_manual_spoke_advertise_cidrs"] = bgpMSAN
 
-			if len(gw.ExcludeCidrList) != 0 {
-				transitGateway["excluded_advertised_spoke_routes"] = strings.Join(gw.ExcludeCidrList, ",")
-			} else {
-				transitGateway["excluded_advertised_spoke_routes"] = ""
+		if goaviatrix.IsCloudType(gw.CloudType, goaviatrix.AWSRelatedCloudTypes|goaviatrix.AzureArmRelatedCloudTypes) {
+			tags := &goaviatrix.Tags{
+				ResourceType: "gw",
+				ResourceName: transitGateway["gw_name"].(string),
+				CloudType:    gw.CloudType,
 			}
 
-			gwDetail, err := client.GetGatewayDetail(gw)
+			tagList, err := client.GetTags(tags)
 			if err != nil {
-				return fmt.Errorf("couldn't get Aviatrix Transit Gateway: %s", err)
+				log.Printf("[WARN] Failed to get tags for transit gateway %s: %v", tags.ResourceName, err)
 			}
 
-			transitGateway["enable_firenet"] = gwDetail.EnableFireNet
-			transitGateway["enable_transit_firenet"] = gwDetail.EnableTransitFireNet
-			transitGateway["enable_egress_transit_firenet"] = gwDetail.EnableEgressTransitFireNet
-			transitGateway["customized_transit_vpc_routes"] = gwDetail.CustomizedTransitVpcRoutes
+			if len(tagList) > 0 {
+				transitGateway["tag_list"] = tagList
```

Comment:
`tag_list` is deprecated, you should be using the `tags` attribute from Transit Gateway. On a separate note, I feel like we should not be calling any other APIs in this data source. Since tags information is not given by default, it should not be part of this resource.